syntax = "proto3";

service Consensus {
    rpc Upload_batch (Batch) returns (Result);
    rpc Train (Training_request) returns (Result);
    rpc Fetch_training_result (Fetch_training_result_request) returns (Result);
}

message Batch {
    string batch_id = 1;
    string data = 2;
}

message Training_request {
    repeated string fuel = 1;
    repeated string furnace = 2;
}

message Fetch_training_result_request {
    string trainingId = 1;
}

message Result {
    enum StatusCode {
        SUCCESS = 0;
        INVALID_REQUEST = 1;
        CONSENSUS_FAILED = 2;
    }
    StatusCode code = 1;
    string description = 2;
}

